test/features/pagecontent/rerendering.js

```diff
@@ -6,46 +6,150 @@
 var preq   = require('preq');
 var server = require('../../utils/server.js');
+var P = require('bluebird');
 
 describe('page re-rendering', function () {
     this.timeout(20000);
 
     before(function () { return server.start(); });
 
-    var r1 = '615503804';
-    var r2 = '615503846';
-
+    // A test page that includes the current date, so that it changes if
+    // re-rendered more than a second apart.
+    var dynamic1 = '/html/User:GWicke%2fDate/653530930';
+    var dynamic2 = '/html/User:GWicke%2fDate/653529842';
 
     function hasTextContentType(res) {
         assert.contentType(res, 'text/html');
     }
 
-
     it('should render & re-render independent revisions', function () {
         var r1tid1;
         var r1tid2;
         var r2tid1;
         return preq.get({
-            uri: server.config.bucketURL + '/html/Main_Page/' + r1
+            uri: server.config.bucketURL + dynamic1
         })
         .then(function (res) {
             assert.deepEqual(res.status, 200);
             r1tid1 = res.headers.etag;
             hasTextContentType(res);
 
+            // delay for 1s to make sure that the timestamp differs on re-render
+            return P.delay(1500)
+            .then(function() {
+                return preq.get({
+                    uri: server.config.bucketURL + dynamic1,
+                    headers: { 'cache-control': 'no-cache' }
+                });
+            });
+        })
+        .then(function (res) {
+            // Since this is a dynamic page which should render the same each
+            // time, the tid should not change.
+            r1tid2 = res.headers.etag;
+            assert.notDeepEqual(r1tid2, r1tid1);
+            assert.notDeepEqual(r1tid2, undefined);
+            hasTextContentType(res);
+
+            return preq.get({
+                uri: server.config.bucketURL + dynamic1 + '/' + r1tid1
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r1tid1);
+            hasTextContentType(res);
+
+            return preq.get({
+                uri: server.config.bucketURL + dynamic1
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r1tid2);
+            hasTextContentType(res);
+
             return preq.get({
-                uri: server.config.bucketURL + '/html/Main_Page/' + r1,
+                uri: server.config.bucketURL + dynamic1 + '/' + r1tid2
+            });
+        })
+        .then(function (res) {
+            assert.deepEqual(res.headers.etag, r1tid2);
+            hasTextContentType(res);
+
+            return preq.get({
+                uri: server.config.bucketURL + dynamic2
+            });
+        })
+        .then(function (res) {
+            r2tid1 = res.headers.etag;
+            assert.deepEqual(res.status, 200);
+            hasTextContentType(res);
+
+            // Delay a bit to give the async save time to complete
```

Comment:
It's not clear what this comment refers to - there is no use of `P.delay()` here, and the URI is the same that's returned in the block. But the comment is actually right - there should be a delay to account for the save, but there is none.